src/store/hotspotDetails/hotspotDetailsSlice.ts

```diff
@@ -72,66 +33,135 @@ const fillMissingHourlyValues = (array: Sum[]) => {
   })
 }
 
-export const fetchHotspotWitnessSums = createAsyncThunk(
-  'hotspotDetails/fetchWitnessSums',
-  async (params: FetchWitnessSumsParams) => {
+const fetchHotspotWitnessSums = async (
+  params: FetchDetailsParams,
+  today: Date,
+  previousMaxDate: Date,
+  bucket: Bucket,
+) => {
+  previousMaxDate.setDate(previousMaxDate.getDate() - params.numDays)
+  const witnessSums = await getHotspotWitnessSums({
+    address: params.address,
+    minTime: `-${params.numDays} day`,
+    maxTime: today,
+    bucket,
+  })
+  const witnessSumsPrevious = await getHotspotWitnessSums({
+    address: params.address,
+    minTime: `-${params.numDays} day`,
+    maxTime: previousMaxDate,
+    bucket,
+  })
+  if (bucket === 'hour') {
+    fillMissingHourlyValues(witnessSums)
+    fillMissingHourlyValues(witnessSumsPrevious)
+  }
+  witnessSums.reverse()
+  const totalAvg = witnessSums.reduce((a, b) => ({ avg: a.avg + b.avg } as any))
+  const prevAvg = witnessSumsPrevious.reduce(
+    (a, b) => ({ avg: a.avg + b.avg } as any),
+  )
+  const witnessAverage = totalAvg.avg / witnessSums.length
+  const witnessAveragePrev = prevAvg.avg / witnessSumsPrevious.length
+  return {
+    witnessSums,
+    witnessAverage,
+    witnessChange: calculatePercentChange(witnessAverage, witnessAveragePrev),
+  }
+}
+
+export const fetchHotspotChallengeSums = async (
+  params: FetchDetailsParams,
+  today: Date,
+  previousMaxDate: Date,
+  bucket: Bucket,
+) => {
+  const challengeSums = await getHotspotChallengeSums({
+    address: params.address,
+    minTime: `-${params.numDays} day`,
+    maxTime: today,
+    bucket,
+  })
+  const challengeSumsPrevious = await getHotspotChallengeSums({
+    address: params.address,
+    minTime: `-${params.numDays} day`,
+    maxTime: previousMaxDate,
+    bucket,
+  })
+  challengeSums.reverse()
+  const totalSum = challengeSums.reduce(
+    (a, b) => ({ sum: a.sum + b.sum } as any),
+  )
+  const prevSum = challengeSumsPrevious.reduce(
+    (a, b) => ({ sum: a.sum + b.sum } as any),
+  )
+  return {
+    challengeSums,
+    challengeSum: totalSum.sum,
+    challengeChange: calculatePercentChange(totalSum.sum, prevSum.sum),
+  }
+}
+
+export const fetchHotspotDetails = createAsyncThunk(
```

Comment:
Decided to combine them all into one call since we were calling them all at the same time anyway. Now we can load them in parallel